uwblueprint/focus-on-nature
# feat: send waitlisted camper registration link

## Notion ticket link

[Camp Overview: Send registration invite link to waitlisted camper](https://www.notion.so/uwblueprintexecs/Camp-Overview-Send-registration-invite-link-to-waitlisted-camper-bc20b40de7684edfaa30771c8a116754)



## Implementation description
* Moved waitlisted camper type to `CamperTypes`
* Added function to update waitlisted camper registration status in `CamperAPIClient`
* Added toast message in waitlist table to display registration invite success/failure

![Screen Shot 2022-09-17 at 1 43 25 PM](https://user-images.githubusercontent.com/46568041/190869703-e71b0b39-fe1a-4db8-b13f-c26c4d6cd409.png)
![Screen Shot 2022-09-17 at 1 43 54 PM](https://user-images.githubusercontent.com/46568041/190869719-2cc0da81-7cf3-4bf8-8b1c-8174b4a00243.png)


## Steps to test
1. go to a camps overview page e.g. http://localhost:3000/admin/camp/630a63450eb59852f9d01b1e
2. go to the waitlist table tab at the bottom of the page 
3. click on the three-dot button and select "send registration form" 

**note - to fully see the feature you would need to change the waitlisted camper email to your own email in the database e.g. https://cloud.mongodb.com/v2/61ebbfc0866f7f3a496b1389#metrics/replicaSet/61ebc0e00a9a4c02dfab887f/explorer/focusOnNature/waitlistedcampers/find


## What should reviewers focus on?
* toast message styling 
* feature functionality

## Checklist
